BoostRunClub/Views/ActivityScene/ActivityCellView.swift

```diff
@@ -23,11 +23,23 @@ class ActivityCellView: UITableViewCell {
         commonInit()
     }
 
-    init() {
-        super.init(style: .default, reuseIdentifier: String(describing: Self.self))
+    override init(frame: CGRect = .zero) {
+        super.init(frame: frame)
         commonInit()
     }
 
+    override func preferredLayoutAttributesFitting(
+        _ layoutAttributes: UICollectionViewLayoutAttributes
+    ) -> UICollectionViewLayoutAttributes {
+        setNeedsLayout()
+        layoutIfNeeded()
```

Comment:
>만일 main run loop에서 하나의 View가 setNeedsLayout을 호출하고 그 다음 layoutIfNeeded를 호출한다면 layoutIfNeeded는 그 즉시 View의 값이 재계산되고 화면에 반영하기 때문에 setNeedsLayout이 예약한 layoutSubviews 메소드는 update cycle에서 반영해야할 변경된 값이 존재하지 않기 때문에 호출되지 않습니다.

>이러한 동작 원리로 layoutIfNeeded는 그 즉시 값이 변경되어야 하는 애니매이션에서 많이 사용됩니다. 만일 setNeedsLayout을 사용한다면 애니매이션 블록에서 그 즉시 View의 값이 변경되는 것이 아니라 추후 update cycle에서 값이 반영되므로 값의 변경은 이루어지지만 애니매이션 효과는 볼 수 없는 것입니다.
setNeedsLayout과 layoutIfNeeded의 차이점은 동기적으로 동작하느냐 비동기적으로 동작하느냐의 차이입니다.

`setNeedsLayout`을 호출하고 `layoutIfNeeded`를 호출하면,`layoutIfNeeded`가 즉시 `layoutSubviews`를 발동시키기 때문에 `setNeedsLayout`으로 예약한 `layoutSubviews`는 호출되지 않는데, 혹시 이 두가지를 함께 적으신게 어떤 이유가 있는지 궁금합니다. 🙂

[내용출처](https://baked-corn.tistory.com/105)

Reply:
layout을 수행할 때 그 안에 어떠한 flag를 두어서
이 flag가 설정되어있을 때만 layout 하도록 하는 걸로 알고 있습니다.

그래서 `layoutIfNeeded()` 는 이 flag 의 상태에 따라 레이아웃이 될 수도 안될 수도 있다고 알고있습니다.
따라서 확실하게 layout해주기 위해 `setNeedsLayout`을 호출해 이 flag를 설정하고
`layoutIfNeeded`를 호출해 주었습니다.

Reply:
흠 그렇군요. 저도 그 flag에 대해 들어본적이 있어서 찾아보는 중인데 잘 못찾겠네요. 혹시 해당 내용에 대해 확인할 수 있는 레퍼런스 같은게 있을까요? 
직접 찾아보니 대부분의 내용이 setNeedsLayout은 다음 update cycle에 layoutSubviews를 반영하도록 하고 layoutIfNeeded는 즉시 반영한다는 차이점만 담고 있네요.

Reply:
[출처](https://programmersought.com/article/92151714289/)
> layoutIfNeeded
layoutIfNeeded allows the layout refresh to call the layoutSubviews method immediately before the drawing cycle loop reaches the node. In other words, **calling layoutIfNeeded will immediately re-layout, provided the object layout settings have changed or the setNeedsLayout method flag was called before,** otherwise layoutIfNeeded will not trigger the call to layoutSubviews.

관련 내용입니다

Reply:
감사합니다. 한번 읽어볼게요. 이 PR은 머지하도록 하겠습니다.